components_library/processors/cpu_types.py

```diff
@@ -32,3 +32,6 @@ class CPUTypes(Enum):
     KVM = 2
     O3 = 3
     TIMING = 4
+    LINGEN = 5
+    RNDGEN = 6
+    CMXGEN = 7
```

Comment:
I am strongly against this kind of change. We have to find a better way.

Reply:
Is it necessary to specify each of the random cores in this way? Could we not have something like "MISC" core for all these? A kinda "catch-all" for cases like this?

Alternatively we can abstract our core hierarchy a bit more:

"AbstractNormalCore" (accepts CPU type as an argument) and "AbstractSpecialCore" (does not have a CPU type), both of which inherit from "AbstractCore".

Another possibility: Multiple inheritance. SimpleCores inherit from both "AbstractCore" and "gem5Core", gem5Core is the one that enforces the CPUTypes.

Reply:
Can we just drop this Enum? I really don't like having to specify the core types this way... Why is it needed?

Reply:
Right now the user needs to specify the CPU type of the SimpleProcessor. The alternative to this would be to have a SimpleProcessor for each CPU type: SimpleTimingProcessor, SimpleAtomicProcessor, etc, which i think would be a bit funny.

The swappableProcessor (see the parsec_disk_run.py) also makes the user select what type of core to swap to and from.

Reply:
I still don't like it. What if I wanted to have an out of order core that is 4-wide? What about an in-order core with 4 ALUs?

I think we should probably go the core route that Mahyar did and instead of specifying a *core type* give a class/factory for creating cores.

I still like the philosophy of extensible, not configurable. :)